src/components/pages/coach/CoachDashboard/index.jsx

```diff
@@ -24,11 +38,508 @@ import { ReactComponent as MorningImg } from  '../../../../assets/icons/morning.
+        clientOnlyCache.query({
+            query: GET_DATA,
+            variables: { "token": localStorage.getItem("fprint") }
+        }).then(({data}) => {
+            console.log("Fetched data from cache.");
+            if(data.userSelf){
+                this.setState({
+                    coachusers: {
+                        userdata: data.userSelf
+                    }
+                }, () => this._fetchUsersWithoutCache());
+            }
+        })
+        .catch(error => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        })
+    }
+    
+    _fetchAllUsers = (customers) => {
+        clientNoCache.query({
+            query: GET_USERS_ALL,
+            variables: { "token": localStorage.getItem("fprint") }
+        }).then(({data}) => {
+            let users = data.userAll;
+            this._getDifference(users, customers);
+        })
+        .catch(error => {
+            this.setState({
+                syncing: false
+            }, () => console.log("Error",error));
+        })
+    }
+
+    _getDifference = (users, customers) => {
+        // Check if user is in Shopify but not in Charm
+        let selection = customers.map((customer, i) => {
+            if(users.some(user => user.telephone === customer.phone)){
+                // User already exists
+                return undefined;
+            } else{
+                // User does not exist
+                return customer;
+            }
+        })
+
+        // Remove all undefined entries from array
+        selection = selection.filter(function(item){
+            return typeof item !== 'undefined';  
+        });
+
+        selection.map((customer, i) => {
+            if(customer){
+                this._createUser(customer);
+            }
+        });
+
+        this._checkForUpdates(users, customers);
+    }
+
+    _checkForUpdates = (users, customers) => {
+        console.log(users);
+        console.log(customers);
+        let t = 0;
+        users.map((user, u) => {
+            customers.map((customer, c) => {
+                if(user.telephone === customer.phone){
+                    // Synced user counter +1
+                    t++;
+                    if(
+                        user.email === customer.email &&
+                        user.firstName === customer.first_name &&
+                        user.lastName === customer.last_name &&
+                        user.address === customer.default_address.address1 &&
+                        user.city === customer.default_address.city &&
+                        user.postalCode === customer.default_address.zip &&
+                        user.country === customer.default_address.country_code &&
+                        user.telephone === customer.phone &&
+                        user.newsletter === customer.accepts_marketing
+                    ){
+                        console.log("Data good");
```

Comment:
# please remove `console` (no-console)

In JavaScript that is designed to be executed in the browser, it's considered a best practice to avoid using methods on `console`. Such messages are considered to be for debugging purposes and therefore not suitable to ship to the client. In general, calls using `console` should be stripped before being pushed to production.

```js
console.log("Made it here.");
console.error("That shouldn't have happened.");
```

Examples of **incorrect** code for this rule:

```js
/* eslint no-console: "error" */

console.log("Log a debug level message.");
console.warn("Log a warn level message.");
console.error("Log an error level message.");
```

Reply:
Noted.
However, the console functions do not exist without reason. It is a really great, fully integrated development tool.